applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_custom_scipy_solver.py

```diff
@@ -0,0 +1,147 @@
+                                                              builder_and_solver,
+                                                              False,
+                                                              False,
+                                                              False,
+                                                              False )
+
+    def _MassMatrixComputation(self): 
+        self.GetComputingModelPart().ProcessInfo.SetValue(StructuralMechanicsApplication.BUILD_LEVEL,1) #Mass Matrix
+        scheme = self.get_mechanical_solution_strategy().GetScheme() 
+
+        aux = self.get_mechanical_solution_strategy().GetSystemMatrix() 
+        self.space.SetToZeroMatrix(aux)
+       
+        # Create dummy vectors
+        b = self.space.CreateEmptyVectorPointer()
+        self.space.ResizeVector( b, self.space.Size1(aux) )
+        self.space.SetToZeroVector(b)
+
+        xD = self.space.CreateEmptyVectorPointer()
+        self.space.ResizeVector( xD, self.space.Size1(aux) )
+        self.space.SetToZeroVector(xD)
+
+        # Build matrix
+        builder_and_solver = self.get_builder_and_solver()
+        builder_and_solver.Build(scheme, self.GetComputingModelPart(), aux, b)
+        # Apply Constraints
+        builder_and_solver.ApplyConstraints(scheme, self.GetComputingModelPart(), aux, b)
+        # Apply Boundary Conditions
+        builder_and_solver.ApplyDirichletConditions(scheme, self.GetComputingModelPart(), aux, xD, b)
+        # Convert Mass matrix to scipy
+        M = KratosMultiphysics.scipy_conversion_tools.to_csr(aux)
+        return M
+
+    def _StiffnessMatrixComputation(self): 
+        self.GetComputingModelPart().ProcessInfo.SetValue(StructuralMechanicsApplication.BUILD_LEVEL,2) #Stiffness Matrix
+        scheme = self.get_mechanical_solution_strategy().GetScheme() 
+        
+        aux = self.get_mechanical_solution_strategy().GetSystemMatrix() 
+        self.space.SetToZeroMatrix(aux)
+
+        # Create dummy vectors
+        b = self.space.CreateEmptyVectorPointer()
+        self.space.ResizeVector( b, self.space.Size1(aux) )
+        self.space.SetToZeroVector(b)
+
+        xD = self.space.CreateEmptyVectorPointer()
+        self.space.ResizeVector( xD, self.space.Size1(aux) )
+        self.space.SetToZeroVector(xD)
+
+        # Build matrix
+        builder_and_solver = self.get_builder_and_solver()
+        builder_and_solver.Build(scheme, self.GetComputingModelPart(), aux, b)
+        # Apply constraints
+        builder_and_solver.ApplyConstraints(scheme, self.GetComputingModelPart(), aux, b)
+        # Apply boundary conditions
+        builder_and_solver.ApplyDirichletConditions(scheme, self.GetComputingModelPart(), aux, xD, b)
+        # Convert stiffness matrix to scipy
+        K = KratosMultiphysics.scipy_conversion_tools.to_csr(aux)
+
+        return K
+        
+    def SolveSolutionStep(self):
+        ## Obtain scipy matrices
+        # M = self._MassMatrixComputation()
```

Comment:
why commented?

Reply:
Because that function should be overridden in the derived class. I just wanted to show how it works. Shall i uncomment it?

Reply:
If this function should always be overridden and never called directly, raise an error here
```python
        raise Exception("This function has to be implemented in the derived class")
```
The way it could be implemented could be explained in a docstring for this function, using the `"""` see e.g. kratos/python.

Reply:
I would call it only if you have a dynamic scheme

Reply:
this requires a special scheme...which is designed on purpose to compute the mass matrix